src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts

```diff
@@ -148,34 +141,22 @@ export class GithubAppsLogsComponent implements OnInit, AfterViewInit {
           tap(() => {
             this.sortDirection = this.sort.direction;
             this.pageFilter = this.filter.nativeElement.value;
-            this.paginator.pageIndex = 0;
+            this.paginator.pageIndex = 0; //go back to first page after adding filter
           }),
           takeUntil(this.ngUnsubscribe)
         )
         .subscribe(() => {
-          this.loadEvent(
-            this.matDialogData,
-            this.paginator.pageIndex * this.paginator.pageSize,
-            this.paginator.pageSize,
-            this.pageFilter,
-            this.sort.direction,
-            this.sortCol
-          );
+          this.loadEvents(this.paginator.pageIndex, this.paginator.pageSize, this.pageFilter, this.sort.direction, this.sortCol);
         });
     });
   }
 
-  loadEvent(
-    matDialogData: { userId?: number; organization?: string },
-    pageIndex: number,
-    pageSize: number,
-    filter: string,
-    sortDirection: string,
-    sortCol: string
-  ) {
-    const filtered = filter !== null && filter !== undefined && filter.length > 0;
+  loadEvents(pageIndex: number, pageSize: number, filter: string, sortDirection: string, sortCol: string) {
+    const filtered = filter?.length > 0;
     let lambdaEvents: Observable<HttpResponse<LambdaEvent[]>>;
-    if (matDialogData.userId) {
+    this.loading = true;
+    this.dataSource.data = [];
```

Comment:
Lines 157 and 158 now cause a big flashing when paging through the dialog.

Sorry, I know I suggested you might need to move `loading`, but it's causing the bulk of the component to disappear every time you page. So maybe you should do it only once after all, although you won't have an indicator as you're paging.

Line 158 is emptying the table before it's fetched the new content, so you see the table contents shrink then grow. You might only want to set to an empty array if the request fails.